stdlib/public/Synchronization/Mutex/LinuxImpl.swift

```diff
@@ -0,0 +1,137 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Atomics open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SynchronizationShims
+
+@_alwaysEmitIntoClient
+@_extern(c, "gettid")
```

Comment:
I know when I did the `Threading` code that I found that older Glibc didn't have a wrapper for `gettid()`. Worth testing to make sure this builds on whatever the current oldest thing we support is.

Reply:
Yeah, glibc only added this in 2.30 which was released in 2019. This is recent enough to warrant just directly invoking the syscall.